src/components/Filters/index.tsx

```diff
@@ -243,24 +247,51 @@ class Filters extends React.Component<FiltersProps, FiltersState> {
 		return update;
 	}
 
-	handleEditChange(value: string, attribute: keyof FilterRule) {
-		let update = this.state.edit;
+	updateFilter(
+		filterBase: FilterModel,
+		attribute: keyof FilterModel,
+		value: any,
+	) {
+		let filter = { ...filterBase };
 
-		if (attribute === 'name' && update.name !== value) {
-			update = this.setDefaultEditData(update, value);
+		if (attribute === 'name' && filter.name !== value) {
+			filter = this.setDefaultEditData(filter, value) as FilterModel;
```

Comment:
Preexisting nit: that method should really be called `getDefaultEditData`, right? It doesn't mutate anything afaict, it just gives you the data.

Reply:
Good point!